app/src/main/java/com/pilot51/voicenotify/AppListScreen.kt

```diff
@@ -15,212 +15,384 @@
+                            it,
+                            hasOverride,
+                            toggleIgnore,
+                            onConfigureApp,
+                            onRemoveOverrides
+                        )
+                    }
+                }
+            }
+        }
+        val yOffset = alphabetRelativeDragYOffset
+        if (yOffset != null && currentLetter != null) {
+            ScrollingBubble(
+                boxConstraintMaxWidth = this.maxWidth,
+                bubbleOffsetYFloat = yOffset + alphabetDistanceFromTopOfScreen,
+                currAlphabetScrolledOn = currentLetter,
+            )
+        }
+    }
 }
 
+@Composable
+fun PackageImage(context: Context, packageName: String, modifier: Modifier = Modifier) {
+    val packageManager: PackageManager = context.packageManager
+    val result = runCatching {
+        val appInfo = packageManager.getApplicationInfo(packageName, 0)
+        val iconBitmap = appInfo.loadIcon(packageManager).toBitmap().asImageBitmap()
+        Image(
+            painter = BitmapPainter(iconBitmap),
+            contentDescription = null,
+            modifier = modifier
+        )
+    }
+    result.onFailure {
+        Image(
+            painter = painterResource(R.drawable.ic_launcher_foreground),
+            contentDescription = null,
+            modifier = modifier
+        )
+    }
+}
+
+
 @Composable
 private fun AppListItem(
-	app: App,
-	hasOverride: Boolean,
-	toggleIgnore: (app: App) -> Unit,
-	onConfigureApp: (app: App) -> Unit,
-	onRemoveOverrides: (app: App) -> Unit
+    app: App,
+    hasOverride: Boolean,
+    toggleIgnore: (app: App) -> Unit,
+    onConfigureApp: (app: App) -> Unit,
+    onRemoveOverrides: (app: App) -> Unit
 ) {
-	var showRemoveOverridesDialog by remember { mutableStateOf(false) }
-	ListItem(
-		modifier = Modifier.clickable {
-			onConfigureApp(app)
-		},
-		headlineContent = {
-			Text(
-				text = app.label,
-				fontSize = 24.sp
-			)
-		},
-		supportingContent = {
-			Text(app.packageName)
-		},
-		trailingContent = {
-			Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
-				if (hasOverride) {
-					IconButton(onClick = { showRemoveOverridesDialog = true }) {
-						Icon(
-							imageVector = Icons.Outlined.Cancel,
-							contentDescription = stringResource(R.string.remove_app_overrides)
-						)
-					}
-				}
-				Checkbox(
-					checked = app.enabled,
-					modifier = Modifier.focusable(false),
-					onCheckedChange = { toggleIgnore(app) }
-				)
-			}
-		}
-	)
-	if (showRemoveOverridesDialog) {
-		ConfirmDialog(
-			text = stringResource(R.string.remove_app_overrides_confirm, app.label),
-			onConfirm = { onRemoveOverrides(app) },
-			onDismiss = { showRemoveOverridesDialog = false }
-		)
-	}
+    var showRemoveOverridesDialog by remember { mutableStateOf(false) }
+    var context = LocalContext.current
+
+    ListItem(
+        modifier = Modifier
+            .toggleable(
```

Comment:
Please replace the `toggleable` modifier with this to restore per-app settings:
```
.clickable {
    onConfigureApp(app)
}
```

Reply:
OK, I review the code agian. LazyAlphabetIndexRow also has bug